.github/workflows/delete-merged-branches.yml

```diff
@@ -0,0 +1,157 @@
+---
+name: 'Delete Merged Branches'
+
+# This workflow automatically deletes branches that are associated with
+# closed or merged pull requests. It runs weekly on Sundays at midnight UTC.
+#
+# Protected branches (will NOT be deleted):
+# - main
+# - master
+# - beta
+# - otel-sdk
+# - release* (case-insensitive: release-1.0, release/v2.0, etc.)
+# - Release* (case-insensitive: Release-1.0, Release/v2.0, etc.)
+# - legacy-* (e.g., legacy-v1, legacy-old)
+# - *-legacy (e.g., old-legacy, v1-legacy)
+#
+# The workflow will only delete branches that:
+# 1. Have associated pull requests
+# 2. All PRs for the branch are either merged or closed (no open PRs)
+# 3. Do not match any protected branch pattern
+
+on:
+  schedule:
+    # Run every Sunday at midnight UTC
+    - cron: '0 0 * * 0'
+  workflow_dispatch:
+
+permissions:
+  contents: write
```

Comment:
check the required permissions and add comments with links to support the setting of the permissions